test-framework/vault/pom.xml

```diff
@@ -14,6 +14,18 @@
     <artifactId>quarkus-test-vault</artifactId>
     <name>Quarkus - Test Framework - Vault</name>
 
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>io.quarkus</groupId>
+                <artifactId>quarkus-bom-test</artifactId>
+                <version>${project.version}</version>
+                <type>pom</type>
+                <scope>import</scope>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
```

Comment:
Maybe I'm wrong but shouldn't it be included in the build-parent? @aloubyansky WDYT?

Reply:
The current proposal is to have three different BOMs: runtime, deployment and test. 

1. Runtime BOM manages the dependencies required by runtime modules and nothing else
2. Deployment BOM imports the runtime BOM and manage the dependencies required by deployment modules
3. The test BOM manages the dependencies required by itests unless they are managed by the runtime BOM. The test BOM could perhaps import 1, because all integration tests will most likely need both.

Now importing the test bom in build-parent seems to be wrong, IMO, because it gives Quarkus developers quite an improper impression about what deps are allowed to use where. They might think that having dependency managed in the test BOM is fine in a runtime or deployment module, which it is definitely not.

(Actually the same argument could be used against importing both runtime and deployment BOM in build-parent: it is wrong to have a `-deployment` dependency in a runtime module, although it would work with the current setup. So build-parent should rather import only the runtime BOM and the deployment modules should import the deployment BOM individually.)

To save some POM boiler plate, we have a `parent-it` in Camel Quarkus that imports the test BOM and inherits from `build-parent` (that in turn imports just the runtime BOM): https://github.com/apache/camel-quarkus/blob/master/poms/build-parent-it/pom.xml

Reply:
That makes sense.